pkg/driver/driver_alicloud.go

```diff
@@ -0,0 +1,287 @@
+		request.SystemDiskCategory = c.AlicloudMachineClass.Spec.SystemDisk.Category
+		request.SystemDiskSize = requests.NewInteger(c.AlicloudMachineClass.Spec.SystemDisk.Size)
+	}
+
+	if c.AlicloudMachineClass.Spec.Tags != nil {
+		request.Tag1Key = c.AlicloudMachineClass.Spec.Tags.Tag1Key
+		request.Tag1Value = c.AlicloudMachineClass.Spec.Tags.Tag1Value
+		request.Tag2Key = c.AlicloudMachineClass.Spec.Tags.Tag2Key
+		request.Tag2Value = c.AlicloudMachineClass.Spec.Tags.Tag2Value
+		request.Tag3Key = c.AlicloudMachineClass.Spec.Tags.Tag3Key
+		request.Tag3Value = c.AlicloudMachineClass.Spec.Tags.Tag3Value
+		request.Tag4Key = c.AlicloudMachineClass.Spec.Tags.Tag4Key
+		request.Tag4Value = c.AlicloudMachineClass.Spec.Tags.Tag4Value
+		request.Tag5Key = c.AlicloudMachineClass.Spec.Tags.Tag5Key
+		request.Tag5Value = c.AlicloudMachineClass.Spec.Tags.Tag5Value
+	}
+
+	request.InstanceName = c.MachineName
+	request.ClientToken = utils.GetUUIDV4()
+	request.UserData = base64.StdEncoding.EncodeToString([]byte(c.UserData))
+
+	response, err := c.runInstances(client, request)
+	if err != nil {
+		return "", "", err
+	}
+
+	return c.encodeMachineID(c.AlicloudMachineClass.Spec.Region, response.InstanceIdSets.InstanceIdSet[0]), c.MachineName, nil
+}
+
+// Delete method is used to delete an alicloud machine
+func (c *AlicloudDriver) Delete() error {
+	result, err := c.getVMDetails(c.MachineID)
+	if err != nil {
+		return err
+	} else if len(result) == 0 {
+		// No running instance exists with the given machineID
+		glog.V(2).Infof("No VM matching the machineID found on the provider %q", c.MachineID)
+	}
+
+	if result[0].Status != "Running" && result[0].Status != "Stopped" {
+		return errors.New("ec2 instance not in running/stopped state.")
+	}
+
+	machineID := c.decodeMachineID(c.MachineID)
+
+	client, err := c.getEcsClient()
+	if err != nil {
+		return err
+	}
+
+	err = c.deleteInstance(client, machineID)
+	return err
+}
+
+func (c *AlicloudDriver) stopInstance(client *ecs.Client, machineID string) error {
+	request := ecs.CreateStopInstanceRequest()
+	request.InstanceId = machineID
+	request.ConfirmStop = requests.NewBoolean(true)
+	request.ForceStop = requests.NewBoolean(true)
+
+	_, err := client.StopInstance(request)
+
+	return err
+}
+
+func (c *AlicloudDriver) deleteInstance(client *ecs.Client, machineID string) error {
+	request := ecs.CreateDeleteInstanceRequest()
+	request.InstanceId = machineID
+	request.Force = requests.NewBoolean(true)
+
+	_, err := client.DeleteInstance(request)
+	return err
+}
+
+// GetExisting method is used to get machineID for existing Alicloud machine
+func (c *AlicloudDriver) GetExisting() (string, error) {
+	return c.MachineID, nil
+}
+
+func (c *AlicloudDriver) getVMDetails(machineID string) ([]ecs.Instance, error) {
+	searchClusterName := ""
+	searchNodeRole := ""
+
+	if c.AlicloudMachineClass.Spec.Tags != nil {
+		if strings.Contains(c.AlicloudMachineClass.Spec.Tags.Tag1Key, "kubernetes.io/cluster/") {
```

Comment:
Can we please check if any of the tags contains `kubernetes.io/cluster`, current implementation will prevent us from not having it at first position.
https://github.com/gardener/machine-controller-manager/blob/master/pkg/driver/driver_aws.go#L194-L200